pkg/controller/jobframework/setup_test.go

```diff
@@ -134,18 +160,46 @@ func TestSetupControllers(t *testing.T) {
 				t.Fatalf("Failed to setup manager: %v", err)
 			}
 
-			gotError := manager.setupControllers(mgr, logger, tc.opts...)
+			gotError := manager.setupControllers(ctx, mgr, logger, tc.opts...)
 			if diff := cmp.Diff(tc.wantError, gotError, cmpopts.EquateErrors()); len(diff) != 0 {
 				t.Errorf("Unexpected error from SetupControllers (-want,+got):\n%s", diff)
 			}
 
-			if diff := cmp.Diff(tc.wantEnabledIntegrations, manager.enabledIntegrations.SortedList()); len(diff) != 0 {
+			if tc.delayedIntegrationTest {
+				// Simulate the delayed addition of RayCluster
+				simulateDelayedIntegration(mgr, tc.delayedGVKs)
+				testDelayedIntegration(&manager, "ray.io/raycluster")
```

Comment:
use a for loop on `tc.delayedGVKs` or something along those lines, instead of the hardcoded string.